pkg/cmd/roachtest/tests/util.go

```diff
@@ -82,15 +82,17 @@ func WaitForUpdatedReplicationReport(ctx context.Context, t test.Test, db *gosql
 	// that the report picks up any new tables or zones.
 	tStart := timeutil.Now()
 	for r := retry.StartWithCtx(ctx, retry.Options{}); r.Next(); {
-		var gen time.Time
+		var count int
+		var gen gosql.NullTime
 		if err := db.QueryRowContext(
-			ctx, `SELECT generated FROM system.reports_meta ORDER BY 1 DESC LIMIT 1`,
-		).Scan(&gen); err != nil {
+			ctx, `SELECT count(*), min(generated) FROM system.reports_meta`,
+		).Scan(&count, &gen); err != nil {
 			if !errors.Is(err, gosql.ErrNoRows) {
 				t.Fatal(err)
 			}
-			// No report generated yet.
-		} else if tStart.Before(gen) {
+			// No report generated yet. There are 3 types of reports. We want to
+			// see a result for all of them.
+		} else if count == 3 && tStart.Before(gen.Time) {
```

Comment:
It's unfortunate that we'll bake this assumption that there are exactly 3 types of reports into this roachtest, but I'm not seeing better options. I don't really understand how we expect users to use this stuff. We reference the "ID of the report that this report entry is part of" in https://www.cockroachlabs.com/docs/stable/query-replication-reports.html#system-reports_meta, but never document which ID maps to which report. These are also unexported symbols in the code. Am I missing something?

Reply:
They're currently unexported constants with an unexported type. It could be refactored to be better. I'd rather leave that for later. I'll file an issue about it if you're okay with that.

Reply:
An issue sounds fine. Thanks again.

Reply:
https://github.com/cockroachdb/cockroach/issues/80002